### PR TITLE
nixos/doc: documentation angle bracket syntax

### PR DESCRIPTION
**Pull request by tobiasBora**
###### Motivation for this change

Lack of documentation for `<...>` syntax, and add link to https://nixos.org/nixos/options.html. The goal is to partially solve the 3 year old bug https://github.com/NixOS/nix/issues/338.

###### Things done

I just modified `nixos/doc/manual/configuration/customizing-packages.xml` and `nixos/doc/manual/configuration/summary.xml`, and then I ran `make` in `nixos/doc/manual`, and `nix-build nixos/release.nix -A manual.x86_64-linux` at the source. 

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by multun** [comment]
LGTM,would you mind only committing related changes ?

File: nixos/doc/manual/configuration/customizing-packages.xml [hunk 1]
```diff
@@ -17,7 +17,10 @@
  <warning>
   <para>
    Unfortunately, Nixpkgs currently lacks a way to query available
-   configuration options.
+   configuration options from command line. You can however go to
```
Comment (jtojnar) [inline nixos/doc/manual/configuration/customizing-packages.xml:20]: These are different options than the section is talking about.

You are describing NixOS module options but this is talking about package options (that is the attribute set you pass as `config` when creating `nixpkgs`, or through [`nixpkgs.config` NixOS option](https://nixos.org/nixos/options.html#nixpkgs.config). There still is not a way to query those as far as I am aware.

And for NixOS options, you can actually use `man configuration.nix` or `nixos-option` commands.

**Review by jtojnar — CHANGES_REQUESTED** [review]
Could you also drop the formatting changes in unrelated files so it is easier to review?

File: nixos/doc/manual/configuration/summary.xml [hunk 2]
```diff
@@ -215,6 +215,17 @@ xlink:href="http://nixos.org/nix/manual/#chap-writing-nix-expressions">Nix
      </entry>
      <entry>Apply a function to every element of a list (evaluates to <literal>[ 2 4 6 ]</literal>)</entry>
     </row>
+    <row>
```
Comment (jtojnar) [inline nixos/doc/manual/configuration/summary.xml:218]: Also this should probably go to basic values group, not the builtin functions.

**Comment by fricklerhandwerk** [comment]
Closing as this is long out of sync with the Nixpkgs manual, and since the [Nix manual now documents the syntax and gives it a proper name](https://nixos.org/manual/nix/unstable/language/constructs/lookup-path) so we can talk about it: lookup path

**Pull request closed without merging**